pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -329,17 +336,22 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 	}
 
 	pipelineMeta, pipelineSpec, err := resources.GetPipelineData(ctx, pr, getPipelineFunc)
-	if err != nil {
+	switch {
+	case errors.Is(err, resolution.ErrorRequestInProgress):
+		message := fmt.Sprintf("PipelineRun %s/%s awaiting remote resource", pr.Namespace, pr.Name)
+		pr.Status.MarkRunning(ReasonResolvingPipelineRef, message)
+		return nil
```

Comment:
I think what is specifically confusing is that the remote resolution is returning an error, but the pipelinerun resolution is not returning an error. For taskruns contained in pipelineruns, if a taskrun is still running, we treat this as status "pending", not an error. It would make more sense to me to handle a ref that is still resolving the same way. would it be possible to treat the ResourceRequest as "pending" rather than an error?

Reply:
Ah I see, so if we updated the signature of `resources.GetPipelineData` to something like this:

```go
pipelineMeta, pipelineSpec, pending, err := resources.GetPipelineData(ctx, pr, getPipelineFunc)
switch {
case pending:
  // The code here to mark the PR as pending
case err != nil:
  // The code here to fail the PipelineRun
default:
  // The code to store the pipeline meta and spec on pr.status
}
```

Does that read clearer than the sentinel check we have here atm?

Reply:
After discussing offline and spending a bit of time trying to rejig this bit of the code I think I'm going to leave it as-is for the time being.

There's scope here for a tidier interface between the reconcilers and pipelineref resolution but the codebase is in a bit of an in-between state because we have resolution logic in (a) the reconcilers themselves (b) the resources subpackage which abstracts some resolution mechanics and (c) the remote package which hides away the built-in OCI bundles resolution (and now the new remoteresolution functionality).

A holistic change to the way this all works makes sense, and is def something we can look at in another pr, but for this one I'm trying to avoid changing how tekton's existing resolution mechanics function to limit blast radius of the new code.

Reply:
SGTM! Would you be able to create an issue or something to describe a refactoring that would make more sense?

Reply:
We do have an issue around this, #3305, so I've added a comment describing the remaining work to do here: https://github.com/tektoncd/pipeline/issues/3305#issuecomment-1061755446

Reply:
sounds good, thanks!